Source/GTMSessionFetcher.m

```diff
@@ -3712,18 +3717,44 @@ - (nonnull dispatch_queue_t)callbackQueue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    return _callbackQueue;
+    return _underlyingCallbackQueue;
   }  // @synchronized(self)
 }
 
 - (void)setCallbackQueue:(nullable dispatch_queue_t)queue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    _callbackQueue = queue ?: dispatch_get_main_queue();
+    GTMSESSION_ASSERT_DEBUG(![self isFetchingUnsynchronized],
+                            @"callbackQueue should not change after beginFetch has been invoked");
+
+    _underlyingCallbackQueue = queue ?: dispatch_get_main_queue();
+    _callbackQueue = CreateCallbackQueueWithUnderlyingQueue(_underlyingCallbackQueue);
   }  // @synchronized(self)
 }
 
+static dispatch_queue_t CreateCallbackQueueWithUnderlyingQueue(
+    dispatch_queue_t underlyingCallbackQueue) {
+  const char *underlyingCallbackQueueLabel = dispatch_queue_get_label(underlyingCallbackQueue);
+  NSString *callbackQueueLabel;
+  if (underlyingCallbackQueueLabel[0] == '\0') {
+    callbackQueueLabel = @"GTMSessionFetcher queue targeting callback queue";
+  } else {
+    callbackQueueLabel = [[NSString alloc]
+        initWithFormat:@"GTMSessionFetcher queue targeting `%s`", underlyingCallbackQueueLabel];
+  }
+
+  // Callbacks need to be dispatched serially. For example, the accumulate data block should never
```

Comment:
I think the “why” is important and may not be immediately clear, so I would like to leave that in the comment. That said, I moved the comment to be a documentation comment on the new `serialCallbackQueue` method so that the code flow is not interrupted by this long comment.